pkg/controller/statefulset/stateful_set.go

```diff
@@ -235,70 +218,77 @@ func (psc *StatefulSetController) getPodsForStatefulSet(ps *apps.StatefulSet) ([
 	return result, nil
 }
 
-// getStatefulSetForPod returns the pet set managing the given pod.
-func (psc *StatefulSetController) getStatefulSetForPod(pod *v1.Pod) *apps.StatefulSet {
-	ps, err := psc.psStore.GetPodStatefulSets(pod)
+// getStatefulSetForPod returns the StatefulSet managing the given pod.
+func (ssc *StatefulSetController) getStatefulSetForPod(pod *v1.Pod) *apps.StatefulSet {
+	sets, err := ssc.setStore.GetPodStatefulSets(pod)
 	if err != nil {
 		glog.V(4).Infof("No StatefulSets found for pod %v, StatefulSet controller will avoid syncing", pod.Name)
 		return nil
 	}
-	// Resolve a overlapping statefulset tie by creation timestamp.
-	// Let's hope users don't create overlapping statefulsets.
-	if len(ps) > 1 {
-		glog.Errorf("user error! more than one StatefulSet is selecting pods with labels: %+v", pod.Labels)
-		sort.Sort(overlappingStatefulSets(ps))
+	// More than one set is selecting the same Pod
```

Comment:
THis shouldn't be possible to happen if we use the naming convention in GetPodStatefulSets, right?